### DISCUSSION TITLE
implementing Prosody's mod_client_proxy

### DISCUSSION THREAD

**Discussion by jirkanov** [processone/ejabberd#3779] [Questions]
Hello,

is there a way of achieving what the Prosody module [mod_client_proxy](https://modules.prosody.im/mod_client_proxy.html) does,
i.e. forwarding stanzas from a service to a given client and back?
Or will it require writing a new module? Any tips on that?

My use case is running Jitsi on top of Ejabberd instead of Prosody.

**Comment by badlop** [comment]
I am not aware of any ejabberd module (included in ejabberd, or managed by somebody else) that implements such a feature. I guess you already made a search over github etc, in case there was such an ejabberd module somewhere.

Looking at the mod_client_proxy documentation, such a feature could be implemented as a "Jabber component", that could be used by any jabber server. I mean, you can search also for a server-agnostic implementation of that feature, that could work with ejabberd, and any other jabber server.

Another implementation path would be to write a chatbot that logins to account proxy@yourejabberd.org and performs its proxy tasks there... Maybe there's already some XMPP chatbot with that feature?

For implementing in ejabberd as a component, some modules to look could be:
* mod_echo for the component part
* mod_filter, mod_pottymouth, mod_spam_filter: for the stanza rewrite part

**Comment by licaon-kter** [comment]
> My use case is running Jitsi on top of Ejabberd instead of Prosody.

Since when is that needed? Jitsi Meet used to work okaish with ejabberd as it is.

**Reply by weiss** [reply]
> Since when is that needed? Jitsi Meet used to work okaish with ejabberd as it is.

At some point they modified the Jicofo component to use a normal c2s login, but for some reason it's still addressed as `focus.example.com` rather than `focus@example.com`, so they rely on the address being rewritten.

**Reply by oliviasculley** [reply]
According to https://community.jitsi.org/t/focus-component-logic/100540, it is theoretically possible to get it to use the normal login, it just isn't the default in the docker env-var-generated configuration. Unfortunately, I still can't get jitsi to work with ejabberd :(

**Comment by Eirikr70** [comment]
Hello, I am trying to set up jitsi-meet on top of ejabberd, with no success for now. Do you have any documentation about how to actually achieve it ?

**Comment by weiss** [comment]
Jitsi Meet no longer works with standard XMPP servers.  They use a Prosody version with custom extensions.

**Reply by Eirikr70** [reply]
Thanks for that reactive answer. Is there any way to support screen sharing in videoconferences with xmpp/ejabberd ? Is it something that might be implemented in a foreseeable future ?
